### PR TITLE
Add snapshot details

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HAC-4576

## Description

Screenshot 2023-08-29 at 7 44 48 PM
Screenshot 2023-08-29 at 7 49 00 PM

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/SnapshotDetails/SnapshotDetailsView.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          },
+          {
+            key: 'pipelineruns',
+            label: 'Pipeline runs',
+            component: <SnapshotOverviewTab snapshot={snapshot} commit={commit} />,
+          },
+          {
+            key: 'logs',
+            label: 'Logs',
+            component: <SnapshotOverviewTab snapshot={snapshot} commit={commit} />,
```
Comment (rottencandy) [inline src/components/SnapshotDetails/SnapshotDetailsView.tsx:152]: Both tabs are using same page?

File: src/components/SnapshotDetails/tabs/SnapshotLogs.tsx
Comment (rottencandy) [inline src/components/SnapshotDetails/tabs/SnapshotLogs.tsx:1]: Missed empty file

File: src/components/SnapshotDetails/SnapshotDetailsView.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            label: 'Edit',
+            component: (
+              <Link
+                to={`/application-pipeline/workspaces/${workspace}/applications/${applicationName}/integrationtests/${snapshotName}/edit`}
+              >
+                Edit
+              </Link>
+            ),
+            isDisabled: !canUpdateIntegrationTest,
+            disabledTooltip: "You don't have access to edit this integration test",
```
Comment (rottencandy) [inline src/components/SnapshotDetails/SnapshotDetailsView.tsx:127]:
```suggestion
                to={`/application-pipeline/workspaces/${workspace}/applications/${applicationName}/snapshots/${snapshotName}/edit`}
              >
                Edit
              </Link>
            ),
            isDisabled: !canUpdateSnapshot,
            disabledTooltip: "You don't have access to edit this snapshot",
```

Comment (abhinandan13jan) [reply src/components/SnapshotDetails/SnapshotDetailsView.tsx]: updated

File: config/remotePlugin.js [hunk 3]
```diff
@@ -435,6 +435,21 @@ const routeExtensions = [
...
+      exact: true,
+      component: {
+        $codeRef: 'SnapshotDetailsPage',
+      },
+    },
+    flags: {
+      required: ['HACBS', 'SIGNUP'],
+    },
+  },
+
```
Comment (rottencandy) [inline config/remotePlugin.js:452]: I think we need another section for activeTab, clicking on pipelineruns tab gives 404.

```js
{
    type: 'core.page/route',
    properties: {
      path: '/application-pipeline/workspaces/:workspaceName/applications/:appName/snapshots/:snapshotName/:activeTab',
      exact: true,
      component: {
        $codeRef: 'SnapshotDetailsPage',
      },
    },
    flags: {
      required: ['HACBS', 'SIGNUP'],
    },
  },
```

File: src/components/SnapshotDetails/SnapshotDetailsView.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            label: 'Edit',
+            component: (
+              <Link
+                to={`/application-pipeline/workspaces/${workspace}/applications/${applicationName}/snapshots/${snapshotName}/edit`}
+              >
+                Edit
+              </Link>
+            ),
+            isDisabled: !canUpdateSnapshots,
+            disabledTooltip: "You don't have access to edit this integration test",
```
Comment (rottencandy) [inline src/components/SnapshotDetails/SnapshotDetailsView.tsx:127]:
```suggestion
            disabledTooltip: "You don't have access to edit this snapshot",
```

File: src/components/SnapshotDetails/tabs/SnapshotOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  return (
+    <>
+      <Title headingLevel="h4" className="pf-c-title pf-u-mt-lg pf-u-mb-lg" size="lg">
+        Snapshot details
+      </Title>
+      <Flex>
+        <Flex flex={{ default: 'flex_3' }}>
+          <FlexItem>
+            <DescriptionList
+              data-test="integration-test-details"
```
Comment (rottencandy) [inline src/components/SnapshotDetails/tabs/SnapshotOverview.tsx:76]:
```suggestion
              data-test="snapshot-details"
```

File: src/components/SnapshotDetails/tabs/SnapshotOverview.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                </DescriptionListGroup>
+              )}
+            </DescriptionList>
+          </FlexItem>
+        </Flex>
+
+        <Flex flex={{ default: 'flex_3' }}>
+          <FlexItem>
+            <DescriptionList
+              data-test="integration-test-details"
```
Comment (rottencandy) [inline src/components/SnapshotDetails/tabs/SnapshotOverview.tsx:107]:
```suggestion
              data-test="snapshot-details"
```

File: src/components/SnapshotDetails/tabs/SnapshotOverview.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                <DescriptionListDescription>
+                  <Timestamp timestamp={snapshot.metadata.creationTimestamp ?? '-'} />
+                </DescriptionListDescription>
+              </DescriptionListGroup>
+              {commit && (
+                <DescriptionListGroup>
+                  <DescriptionListTerm>Triggered by</DescriptionListTerm>
+                  <DescriptionListDescription data-test="snapshot-commit-link">
+                    <Link
+                      to={`/application-pipeline/workspaces/${workspace}/applications/${snapshot.spec.application}/commits/${commit.sha}`}
```
Comment (rottencandy) [inline src/components/SnapshotDetails/tabs/SnapshotOverview.tsx:92]: This is giving 404
```suggestion
                      to={`/application-pipeline/workspaces/${workspace}/applications/${snapshot.spec.application}/commit/${commit.sha}`}
```

File: src/components/SnapshotDetails/tabs/SnapshotComponentsListRow.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+const SnapshotComponentsListRow: React.FC<RowFunctionArgs<SnapshotComponentTableData>> = ({
+  obj,
+}) => {
+  const { workspace } = useWorkspaceInfo();
+  return (
+    <>
+      <TableData data-test="snapshot-component-list-row" className={commitsTableColumnClasses.name}>
+        <Link
+          to={`/application-pipeline/workspaces/${workspace}/applications/${obj.application}/component/${obj.name}`}
```
Comment (rottencandy) [inline src/components/SnapshotDetails/tabs/SnapshotComponentsListRow.tsx:24]: Right now this link gives 404, but that should be handled with #763 right?

Comment (rottencandy) [reply src/components/SnapshotDetails/tabs/SnapshotComponentsListRow.tsx:24]: I noticed that PR is using `/components/:componentName` format, so we might need to update to that

File: src/components/SnapshotDetails/SnapshotDetailsView.tsx [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+                buildPipelineName={buildPipelineName}
+              />
+            ),
+          },
+          {
+            key: 'pipelineruns',
+            label: 'Pipeline runs',
+            component: <SnapshotOverviewTab snapshot={snapshot} commit={commit} />,
+          },
+        ]}
```
Comment (rottencandy) [inline src/components/SnapshotDetails/SnapshotDetailsView.tsx:149]: Both tabs are showing the same page, is it expected?

File: config/remotePlugin.js (hunk 3 above)
Comment (abhinandan13jan) [reply config/remotePlugin.js:452]: It is part of the next PR

File: src/components/SnapshotDetails/SnapshotDetailsView.tsx (hunk 9 above)
Comment (abhinandan13jan) [reply src/components/SnapshotDetails/SnapshotDetailsView.tsx]: removed the PipelineRun tab for now

Comment (abhinandan13jan) [reply src/components/SnapshotDetails/SnapshotDetailsView.tsx]: Screenshot 2023-08-17 at 5 09 06 PM
Screenshot 2023-08-17 at 5 09 14 PM

**Comment by abhinandan13jan** [comment]
@karthikjeeyar Can you please check again

**Review by karthikjeeyar — COMMENTED** [review]
Here are some observations:

1. UXD Design contains list of environments and the deployed status applied to it

Design:
image

PR change contains build pipelinerun status, is this expected ? 

Screenshot 2023-08-24 at 6 46 54 PM




2. ContainerImage column overflows to the next column

Screenshot 2023-08-24 at 6 34 51 PM

File: src/components/SnapshotDetails/SnapshotDetailsView.tsx [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+            component: (
+              <Link
+                to={`/application-pipeline/workspaces/${workspace}/applications/${applicationName}/snapshots/${snapshotName}/edit`}
+              >
+                Edit
+              </Link>
+            ),
+            isDisabled: !canUpdateSnapshots,
+            disabledTooltip: "You don't have access to edit this snapshot",
+          },
```
[the remark this answers is not in the document]
Comment (abhinandan13jan) [reply src/components/SnapshotDetails/SnapshotDetailsView.tsx:177]: removed

**Pull request merged**